Add more explanation on readme about how it works (#91)
meilisearch/gatsby-plugin-meilisearch: README.md

```node
"data": {
  "allSitePage": {
    "nodes": [
      {
        "path": "/404/"
      },
      {
        "path": "/404.html"
      },
      {
        "path": "/"
      }
    ]
  }
}
```

**transformer**

Function that transforms the fetched data before sending it to MeiliSearch.

The format of the data retrieved with the previousy executed graphQL query is not compatible with the format expected by MeiliSearch. Therefore, it should be transformed to a compatible format with the help of the `transformer` function.

```bash
transformer: data =>
  data.allSitePage.nodes.map((node, index) => ({
    id: index,
    ...node,
  })),
```

In this function, we map on `data.allSitePage.nodes` in order to return an array of objects that can be indexed by MeiliSearch. We add an `id` field as MeiliSearch needs it for the indexation. As we don't have any field here that can be used as an `id`, we use the index of the current element in the array.

If you want to learn more about these options (`indexUid`, `query` and `transformer`) see [indexes options](#-indexes)

After filling in those fields, your MeiliSearch configuration should look like this:

```node
plugins: [
  {

>     resolve: 'gatsby-plugin-meilisearch,

Reviewer: Missing ``` ' ``` in ```resolve: 'gatsby-plugin-meilisearch```

    options: {
      host: 'http://localhost:7700',
      apiKey: 'masterKey',
      indexes: [
        {

>           indexUid: 'pages_url'

Reviewer: Missing ``` , ``` after ``` indexUid: 'pages_url' ```

docker run -it --rm -p 7700:7700 getmeili/meilisearch:latest ./meilisearch --master-key=masterKey
```

> ## 🎬 Getting started

Reviewer 2: The getting started should be a step by step process to help the user use the plugin. Explainations should be add the top of the readme. See `https://github.com/meilisearch/docs-searchbar.js` as an example.

> This plugin makes it possible for you to get the content of your Gatsby website, and index it to MeiliSearch.

Reviewer 2: ```suggestion
With this plugin, you can index/add the data used in your [Gatsby application](https://www.gatsbyjs.com/docs/content-and-data/) into MeiliSearch. 

For example, using the same queries you are using to fetch and use data from the  [`mdx-plugin`](https://www.gatsbyjs.com/docs/how-to/routing/mdx-plugins/), the data can also be indexed in MeiliSearch (see query example below).
```

> This plugin is not a scrapper, and therefore you will need to tell it which content it has to retrieve. This is made possible thanks to GraphQL and the queries you'll provide in your `gatsby-config.js` file. To learn more about Gatsby and GraphQL queries, you can [discover how it works](https://www.gatsbyjs.com/docs/conceptual/graphql-concepts/).

Reviewer 2: ```suggestion
The plugin is not a scrapper, and therefore you need to specify which content should be indexed. You are able to do so with the GraphQL queries you provide in your `gatsby-config.js` file. The queries are used against the GraphQL API your Gatbsy app provides. 

To learn more about Gatsby and GraphQL queries, you can [discover how it works](https://www.gatsbyjs.com/docs/conceptual/graphql-concepts/).
```

> Once the content has been retrieved, it is send to your MeiliSearch instance for indexation.

Reviewer 2: ```suggestion
During the build of your Gatsby application, the data is fetched from Gatsby and then, it is indexed in your MeiliSearch instance.
```

If you need tools to integrate a search experience on your app, we also have what you need:

- [docs-searchbar](https://github.com/meilisearch/docs-searchbar.js): a tool to display a searchbar on your website

> - [instant-meilisearch](https://github.com/meilisearch/instant-meilisearch): a UI library that lets you quickly build a search interface in your front-end application

Reviewer 2: Not sure this should be kept in the getting started.

```

If you go to http://0.0.0.0:7700/, you should see our [mini-dashboard](https://github.com/meilisearch/mini-dashboard/) without any index nor documents inside. So let's fill it !

**Add your documents to MeiliSearch**

The `gatsby-plugin-meilisearch` fetches and sends your documents for indexation to MeiliSearch on build, thanks to your graphQL queries. You can try it by triggering a build of your website:

`yarn build` or `npm run build`

The build should succeed, and you should see an information telling you that your content was successfully indexed: `success gatsby-plugin-meilisearch - x.xxxs - Documents added to MeiliSearch`

Go back to http://0.0.0.0:7700/, the data you requested was fetched and indexed to MeiliSearch 🎉

If you need tools to integrate a search experience on your app, we also have what you need:

- [docs-searchbar](https://github.com/meilisearch/docs-searchbar.js): a tool to display a searchbar on your website
- [instant-meilisearch](https://github.com/meilisearch/instant-meilisearch): a UI library that lets you quickly build a search interface in your front-end application

## 🛼 Usage

> 

Reviewer 2: Here you should add a context. For example

```
In the gatsby-config.js file, the MeiliSearch plugin accepts the following parameters:
```

### `indexes` (required)

The `indexes` field in an array of objects, each of them representing an [index](https://docs.meilisearch.com/learn/core_concepts/indexes.html#indexes)

You can have one or multiple `index` objects, which can be useful if you want to index your content in separate indexes.

Each `index` object should contain the following fields:

- `indexUid` (required): Name of your MeiliSearch index. Note that if your index already exists, it will be deleted and recreated
- `transformer` (required): function that transforms the fetched data before sending it to MeiliSearch
- `query` (required): the graphQL query you want to be executed in order to retrieve your documents

### `apiKey` (optional)

The `apiKey` field contains the API key if the MeiliSearch instance is password protected.

Example:

Start a MeiliSearch instance with an API key:
`docker run --rm -it -p 7700:7700 -e MEILI_MASTER_KEY=masterKey getmeili/meilisearch:latest`

Set your API key in your `gatsby-config` file:

```node
options: {
  host: 'http://localhost:7700',
  apiKey: 'masterKey',
  indexes: [
    {
      ...yourIndexOptions
    },
  ],

> },

Reviewer 2: It is an example of which part of the configurations? It should already be running. 
Not sure why it is added before the next parameters.

Also, no need to repeat how to run a MeiliSearch.

## 🎬 Getting started

**Creation of a Gastby project**

If you already have an existing Gatsby project, you can pass this stage and go to the next one.
To create a new Gastby project, simply run this command:

`gastby new`

> Follow the steps, then go to the newly created repository and start Gastby:

Reviewer 2: This part should not be documented by us. You can make a link to the documentation of the install of gatsby. Here we will only say 

> Run your Gatsby app, you should have access to: 
> - `http..`
> 
> If you don't have a Gatsby app yet, please refer to the [Gatsby documentation](#link)

Please improve my suggestion

`yarn develop` or `npm run develop`

You have now access to 2 URL:

- `http://localhost:8000/` where your website runs

> - `http://localhost:8000/___graphql` where you can discover and build graphQL queries

Reviewer 2: This all should be added in `installation` after `run a meilisearch`

**Installation of gatsby-plugin-meilisearch**

Add the plugin to your project's dependencies:

> `yarn add gatsby-plugin-meilisearch` or `npm install gatsby-plugin-meilisearch`

Reviewer 2: Commands should be added in triple quotes by convention. Refer to the other SDK's for examples

Then you must add it to your `gatsby-config.js` configuration file:

```node
module.exports = {
  siteMetadata: {
    siteUrl: "https://www.yourdomain.tld",
    title: "The great Gatsby",
  },
  plugins: [
    {
      resolve: 'gatsby-plugin-meilisearch,
      options: {
        host: 'http://localhost:7700',
        indexes: [
          {
            indexUid: 'the_great_gastby',
            transformer: data =>
              data.allSitePage.edges.map(({ node }, index) => ({
                id: index,
                ...node,
              })),
            query: `
            query MyQuery {
              allSitePage {
                edges {
                  node {
                    componentChunkName
                    internalComponentName
                    path
                  }
                }
              }
            }
            `,
          },
        ],
      },
    },

>   ],

Reviewer 2: This should be added step by step

- Add credentials
```
plugins: [
    {
      resolve: 'gatsby-plugin-meilisearch,
      options: {
        host: 'http://localhost:7700',
        apiKey: "masterkey"
        },
      },
    },
```

- Add a query in `indexes`

Here you add a story line where you use  Use the `md` plugin to create a scenario to follow as your query example does not help if the user did not install the md plugin. 

Also explain what `transformer` does in this context

Reviewer 2: Also, we only need the `plugin` field. Not the whole config file

};
```

**Run a MeiliSearch instance**

If you already have a running MeiliSearch instance, you can replace the `host` field in the plugin option and add a new `apiKey` field with your MeiliSearch credentials.

I you don't, you can quickly start a new MeiliSearch instance by running the following commands:

```bash
docker pull getmeili/meilisearch:latest # Fetch the latest version of MeiliSearch image from Docker Hub
docker run -it --rm -p 7700:7700 getmeili/meilisearch:latest ./meilisearch
```

> If you go to http://0.0.0.0:7700/, you should see our [mini-dashboard](https://github.com/meilisearch/mini-dashboard/) without any index nor documents inside. So let's fill it !

Reviewer 2: This should be before the getting started in the `installation` part. 

See [strapi plugin](https://github.com/meilisearch/strapi-plugin-meilisearch/#-installation) or [docs-searchbar](https://github.com/meilisearch/docs-searchbar.js/)

> **Add your documents to MeiliSearch**

Reviewer 2: This should be called `Build your project` as a step of the getting started

> The `gatsby-plugin-meilisearch` fetches and sends your documents for indexation to MeiliSearch on build, thanks to your graphQL queries. You can try it by triggering a build of your website:

Reviewer 2: ```suggestion
The `gatsby-plugin-meilisearch` fetches and sends your documents for indexation to MeiliSearch on build, using the graphQL queries provided in the `gatsby-config.js` file. 

Build your project:

> `yarn build` or `npm run build`

Reviewer 2: Lets assume the user do not have these script name an call them `gatsby build`. Also by convention they should be in triple quotes 

```
gatsby build
```

> The build should succeed, and you should see an information telling you that your content was successfully indexed: `success gatsby-plugin-meilisearch - x.xxxs - Documents added to MeiliSearch`

Reviewer 2: ```suggestion
After the build, a message in your terminal should confirm that your that your content was successfully indexed: 
`success gatsby-plugin-meilisearch - x.xxxs - Documents added to MeiliSearch`
```

I can not suggest it here but here as well, you should add triple quotes. Single quotes are used to emphasis some technical keywords

Go back to http://0.0.0.0:7700/, the data you requested was fetched and indexed to MeiliSearch 🎉

If you need tools to integrate a search experience on your app, we also have what you need:

- [docs-searchbar](https://github.com/meilisearch/docs-searchbar.js): a tool to display a searchbar on your website

> - [instant-meilisearch](https://github.com/meilisearch/instant-meilisearch): a UI library that lets you quickly build a search interface in your front-end application

Reviewer 2: This can be a section called  `Integrate search components`
